File: app/src/main/java/de/bornholdtlee/defaultproject/injection/modules/UtilsModule.java
package de.bornholdtlee.defaultproject.injection.modules;

import net.hockeyapp.android.utils.ImageUtils;

import org.greenrobot.essentials.io.FileUtils;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import de.bornholdtlee.defaultproject.BaseApplication;
import de.bornholdtlee.defaultproject.utils.NetworkUtils;
import de.bornholdtlee.defaultproject.utils.SharedPreferencesUtils;
import de.bornholdtlee.defaultproject.utils.UiUtils;

@Module
public class UtilsModule {

    @Provides
    @Singleton
    NetworkUtils provideNetworkUtils(BaseApplication context) {
        return new NetworkUtils(context);
    }

    @Provides
    @Singleton
    SharedPreferencesUtils provideSharedPreferencesUtils(BaseApplication context) {
        return new SharedPreferencesUtils(context);
    }

    @Provides
    @Singleton
    UiUtils provideUiUtils() {
        return new UiUtils();
    }

    @Provides
    @Singleton
    ImageUtils provideImageUtils() {
        return new ImageUtils();
    }

    @Provides
    @Singleton
    FileUtils provideFileUtils() {
        return new FileUtils();
    }
//
//    @Provides
//    @Singleton
//    ToastBuilder provideToastBuilder() {
//        return new ToastBuilder();
//    }
//
//    @Provides
//    @Singleton
//    DialogBuilder provideDialogBuilder() {
//        return new DialogBuilder();
//    }

}

File: app/src/main/java/de/bornholdtlee/defaultproject/injection/modules/UtilsModule.kt
package de.bornholdtlee.defaultproject.injection.modules

import dagger.Module
import dagger.Provides
import de.bornholdtlee.defaultproject.base.BaseApplication
import de.bornholdtlee.defaultproject.utils.*
import de.bornholdtlee.defaultproject.viewbuilder.DialogBuilder
import de.bornholdtlee.defaultproject.viewbuilder.ToastBuilder
import javax.inject.Singleton

@Module
open class UtilsModule {

    @Provides
    @Singleton
    fun provideNetworkUtils(context: BaseApplication): NetworkUtils {
        return NetworkUtils(context)
    }

    @Provides
    @Singleton
    open fun provideSharedPreferencesUtils(context: BaseApplication): SharedPreferencesUtils {
        return SharedPreferencesUtils(context)
    }

    @Provides
    @Singleton
    fun provideToastBuilder(): ToastBuilder {
        return ToastBuilder()
    }

    @Provides
    @Singleton
    fun provideUiUtils(): UiUtils {
        return UiUtils()
    }

    @Provides
    @Singleton
    fun provideDialogBuilder(): DialogBuilder {
        return DialogBuilder()
    }

    @Provides
    @Singleton
    fun provideAndroidUtils(): AndroidUtils {
        return AndroidUtils()
    }

    @Provides
    @Singleton
    fun provideStringUtils(): StringUtils {
        return StringUtils()
    }

    @Provides
    @Singleton
    fun provideDateUtils(): DateUtils {
        return DateUtils()
    }

}
